infrastructure/configuration/devops-agents/tools.sh

```diff
@@ -1,72 +1,86 @@
-export DEBIAN_FRONTEND=noninteractive
+#!/bin/bash
+set -euo pipefail
 
-sudo echo 'APT::Acquire::Retries "3";' > /etc/apt/apt.conf.d/80-retries
-sudo echo "APT::Get::Assume-Yes \"true\";" > /etc/apt/apt.conf.d/90assumeyes
+export DEBIAN_FRONTEND=noninteractive
 
-sudo add-apt-repository main
-sudo add-apt-repository restricted
-sudo add-apt-repository universe
-sudo add-apt-repository multiverse
-sudo apt update
+# Set APT options
+sudo bash -c 'echo "APT::Acquire::Retries \"3\";" > /etc/apt/apt.conf.d/80-retries'
+sudo bash -c 'echo "APT::Get::Assume-Yes \"true\";" > /etc/apt/apt.conf.d/90assumeyes'
 
 sudo apt-get clean && apt-get update && apt-get upgrade
-sudo apt-get install -y --no-install-recommends \
-  apt-transport-https \
-  build-essential \
-  ca-certificates \
-  curl \
-  gnupg \
-  jq \
-  libasound2 \
-  libgbm-dev \
-  libgconf-2-4 \
-  libgtk2.0-0 \
-  libgtk-3-0 \
-  libnotify-dev \
-  libnss3 \
-  libxss1 \
-  libxtst6 \
-  lsb-release \
-  software-properties-common \
-  unzip \
-  wget \
-  xauth \
-  xvfb \
-  zip
-
-sudo add-apt-repository ppa:git-core/ppa
-sudo add-apt-repository ppa:deadsnakes/ppa
-
-# Git
-sudo apt install -y --no-install-recommends \
-  git \
-  git-lfs \
-  git-ftp
-
-# Python
-sudo apt install -y --no-install-recommends \
-  python3.7 \
-  python3.7-distutils \
-  python3-pip
-
-# Terraform 1.3.3
-curl -fsSL https://apt.releases.hashicorp.com/gpg | apt-key add -
-sudo apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main"
-sudo apt-get install -y terraform=1.3.3
-
-# Checkov
-python3.7 -m pip install -U checkov
+
+# Source the config file
+CONFIG_FILE="/tmp/config.sh"
+if [ -f "$CONFIG_FILE" ]; then
+  echo "Sourcing $CONFIG_FILE"
+  source "$CONFIG_FILE"
+else
+  echo "config.sh not found" >&2
+  exit 1
+fi
+
+# Debugging output for repositories and packages
+echo "APT_REPOSITORIES: ${APT_REPOSITORIES[*]}"
+echo "COMMON_PACKAGES: ${COMMON_PACKAGES[*]}"
+
+# Function to add APT repository
+add_apt_repository() {
+  if sudo add-apt-repository -y "$1"; then
+    echo "Added repository: $1"
+  else
+    echo "Failed to add repository: $1" >&2
+    exit 1
+  fi
+}
+
+# Function to install packages
+install_packages() {
+  if sudo apt-get install -y --no-install-recommends "$@"; then
+    echo "Installed packages: $@"
+  else
+    echo "Failed to install packages: $@" >&2
+    exit 1
+  fi
+}
+
+# Add repositories
+for repo in "${APT_REPOSITORIES[@]}"; do
+  echo "Adding repository: $repo"
+  add_apt_repository "$repo"
+done
+
+# Install common packages
+install_packages "${COMMON_PACKAGES[@]}"
+
+# Install tfenv
+git clone https://github.com/tfutils/tfenv.git ~/.tfenv
+sudo ln -s ~/.tfenv/bin/* /usr/local/bin
+tfenv install "$TFENV_VERSION"
+tfenv use "$TFENV_VERSION"
```

Comment:
`$TFENV_VERSION` should in the github path rather than use for `install/use` as that would install terraform with that version.

e.g. `git clone --depth 1 --branch $TFENV_VERSION https://github.com/tfutils/tfenv.git ~/.tfenv`
should donwload the version specified as tfenv uses tags for versions (which the --branch option can take).